src/requisitionMerge.js

```diff
@@ -90,6 +160,39 @@ const findMatchedRequisition = ({ code: incomingItemCode }) => ({
   productCode: outgoingItemCode,
 }) => outgoingItemCode === incomingItemCode;
 
+/**
+ * Merges an array of incoming regimen lines with outgoing regimen lines.
+ *
+ * @param  {Array} incomingRegimenLines Regimen lines of the incoming requisition
+ * @param  {Array} outgoingRegimenLines Regimen lines of the outgoing requisition
+ * @return {Array} The merged regimen lines.
+ */
+const requisitionRegimensMerge = (incomingRegimenLines, outgoingRegimenLines) => {
+  const incomingLines = [...incomingRegimenLines.map(getMappedRegimenColumns)];
+  const outgoingLines = [...outgoingRegimenLines];
+
+  const outgoingRegimenCodes = new Set(outgoingLines.map(({ code }) => code));
+  const matchingRegimenCodes = new Set(
```

Comment:
isn't this a set of objects, not just codes

Reply:
8f8fef9e952881b28a06d3053f87028697c67b43.